File: backend/src/main/java/com/team01/project/domain/music/dto/SpotifyTrackResponse.java
package com.team01.project.domain.music.dto;

import java.util.List;
import java.util.stream.Collectors;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@JsonIgnoreProperties(ignoreUnknown = true)
public class SpotifyTrackResponse {
	private String id;
	private String name;

	@JsonProperty("uri")
	private String uri;

	@JsonProperty("artists")
	private List<Artist> artists;

	private Album album;

	@Getter
	@Setter
	@JsonIgnoreProperties(ignoreUnknown = true)
	public static class Album {
		private String name;

		@JsonProperty("release_date")
		private String releaseDate;

		@JsonProperty("images")
		private List<Image> images;
	}

	@Getter
	@Setter
	@JsonIgnoreProperties(ignoreUnknown = true)
	public static class Artist {
		private String id;
		private String name;
	}

	@Getter
	@Setter
	@JsonIgnoreProperties(ignoreUnknown = true)
	public static class Image {
		private String url;
	}

	public String getArtistsAsString() {
		return artists.stream().map(Artist::getName).collect(Collectors.joining(", "));
	}

	public String getArtistsIdAsString() {
		return artists.stream().map(Artist::getId).collect(Collectors.joining(", "));
	}
}

File: backend/src/main/java/com/team01/project/domain/music/dto/SpotifyTrackResponse.kt
package com.team01.project.domain.music.dto

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
data class SpotifyTrackResponse(
    val id: String,
    val name: String,

    @JsonProperty("uri")
    val uri: String,

    @JsonProperty("artists")
    val artists: List<Artist>,

    val album: Album
) {
    fun getArtistsAsString(): String {
        return artists.joinToString(", ") { it.name }
    }

    fun getArtistsIdAsString(): String {
        return artists.joinToString(", ") { it.id }
    }

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Album(
        val name: String,

        @JsonProperty("release_date")
        val releaseDate: String,

        @JsonProperty("images")
        val images: List<Image>
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Artist(
        val id: String,
        val name: String
    )

    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Image(
        val url: String
    )
}
